larky/src/main/resources/vendor/OpenPGP/__init__.star

```diff
@@ -1697,468 +2030,288 @@ def _PublicKeyPacket():
-#         elif self.algorithm == 1:
-#             self.data = Message.parse(zlib.decompress(self.data, -15))
-#         elif self.algorithm == 2:
-#             self.data = Message.parse(zlib.decompress(self.data))
-#         elif self.algorithm == 3:
-#             self.data = Message.parse(bz2.decompress(self.data))
-#         else:
-#             pass  # TODO: error?
-#     self.read = read
-#
-#     def body():
-#         body = pack("!B", self.algorithm)
-#         if self.algorithm == 0:
-#             self.data = self.data.to_bytes()
-#         elif self.algorithm == 1:
-#             compressor = zlib.compressobj(
-#                 zlib.Z_DEFAULT_COMPRESSION, zlib.DEFLATED, -15
-#             )
-#             body += compressor.compress(self.data.to_bytes())
-#             body += compressor.flush()
-#         elif self.algorithm == 2:
-#             body += zlib.compress(self.data.to_bytes())
-#         elif self.algorithm == 3:
-#             body += bz2.compress(self.data.to_bytes())
-#         else:
-#             pass  # TODO: error?
-#         return body
-#     self.body = body
-#
-#     def __iter__():
-#         return iter(self.data)
-#     self.__iter__ = __iter__
-#
-#     def __getitem__(item):
-#         return self.data[item]
-#     self.__getitem__ = __getitem__
-#     return self
-def EncryptedDataPacket():
-    """OpenPGP Symmetrically Encrypted Data packet (tag 9).
-    http://tools.ietf.org/html/rfc4880#section-5.7
+def _class_CompressedDataPacket():
+    """OpenPGP Compressed Data packet (tag 8).
+    http://tools.ietf.org/html/rfc4880#section-5.6
     """
-    cls = _Packet()
-    cls.__name__ = 'EncryptedDataPacket'
-    cls.__mro__.insert(0, EncryptedDataPacket)
-    def __init__():
-        self = cls()
-        self.__name__  = cls.__name__
-        self.__class__ = cls
-        return self
-    self = __init__()
-    return self
 
-def MarkerPacket():
-    """OpenPGP Marker packet (tag 10).
-    http://tools.ietf.org/html/rfc4880#section-5.8
-    """
-    cls = _Packet()
-    cls.__name__ = 'MarkerPacket'
-    cls.__mro__.insert(0, MarkerPacket)
-
-    def __init__():
-        self = cls()
-        self.__name__  = cls.__name__
-        self.__class__ = cls
-        return self
-    self = __init__()
-    return self
+    # http://tools.ietf.org/html/rfc4880#section-9.3
+    algorithms = {0: "Uncompressed", 1: "ZIP", 2: "ZLIB", 3: "BZip2"}
+
+    def read(self):
+        self.algorithm = ord(self.read_byte())
+        self.data = self.read_bytes(self.length)
+        if self.algorithm == 0:
+            self.data = Message.parse(self.data)
+        elif self.algorithm == 1:
+            self.data = Message.parse(zlib.decompress(self.data, -15))
+        elif self.algorithm == 2:
+            self.data = Message.parse(zlib.decompress(self.data))
+        elif self.algorithm == 3:
+            self.data = Message.parse(bz2.decompress(self.data))
+        else:
+            pass  # TODO: error?
```

Comment:
Completing the TODO and providing errors would make debugging possible issues more obvious

Reply:
Copied from the original stuff: https://github.com/mahmoudimus/OpenPGP-Python/blob/fork/src/OpenPGP/__init__.py#L1608-L1638